File: sa-reminder-backend/sa-reminder-frontend/src/api/scheduleAttendanceController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request'

/** 此处后端没有提供注释 POST /schedule/attendance/admin/list/page/vo */
export async function listAttendancePage(
  body: API.ScheduleAttendanceQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePageScheduleAttendanceVO>(
    '/schedule/attendance/admin/list/page/vo',
    {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      data: body,
      ...(options || {}),
    }
  )
}

/** 此处后端没有提供注释 POST /schedule/attendance/admin/updateStatus */
export async function updateAttendanceStatus(
  body: API.ScheduleAttendanceUpdateRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>('/schedule/attendance/admin/updateStatus', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /schedule/attendance/checkIn */
export async function checkIn(
  body: API.ScheduleAttendanceCheckInRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseScheduleAttendanceCheckInVO>('/schedule/attendance/checkIn', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

File: sa-reminder-backend/src/main/java/com/workspace/sareminderbackend/controller/ScheduleAttendanceController.java
package com.workspace.sareminderbackend.controller;

import com.mybatisflex.core.paginate.Page;
import com.workspace.sareminderbackend.annotation.AuthCheck;
import com.workspace.sareminderbackend.common.BaseResponse;
import com.workspace.sareminderbackend.common.ResultUtils;
import com.workspace.sareminderbackend.constant.UserConstant;
import com.workspace.sareminderbackend.exception.ErrorCode;
import com.workspace.sareminderbackend.exception.ThrowUtils;
import com.workspace.sareminderbackend.model.dto.schedule.ScheduleAttendanceCheckInRequest;
import com.workspace.sareminderbackend.model.dto.schedule.ScheduleAttendanceQueryRequest;
import com.workspace.sareminderbackend.model.dto.schedule.ScheduleAttendanceUpdateRequest;
import com.workspace.sareminderbackend.model.entity.User;
import com.workspace.sareminderbackend.model.vo.ScheduleAttendanceCheckInVO;
import com.workspace.sareminderbackend.model.vo.ScheduleAttendanceRateVO;
import com.workspace.sareminderbackend.model.vo.ScheduleAttendanceVO;
import com.workspace.sareminderbackend.service.ScheduleAttendanceService;
import com.workspace.sareminderbackend.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/schedule/attendance")
public class ScheduleAttendanceController {

    @Resource
    private ScheduleAttendanceService scheduleAttendanceService;

    @Resource
    private UserService userService;

    @PostMapping("/checkIn")
    @AuthCheck
    public BaseResponse<ScheduleAttendanceCheckInVO> checkIn(@RequestBody ScheduleAttendanceCheckInRequest request,
                                                             HttpServletRequest httpServletRequest) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(httpServletRequest);
        return ResultUtils.success(scheduleAttendanceService.checkInByTask(request, loginUser));
    }

    @PostMapping("/admin/list/page/vo")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Page<ScheduleAttendanceVO>> listAttendancePage(@RequestBody ScheduleAttendanceQueryRequest request) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        return ResultUtils.success(scheduleAttendanceService.listAttendancePage(request));
    }

    @PostMapping("/admin/updateStatus")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> updateAttendanceStatus(@RequestBody ScheduleAttendanceUpdateRequest request) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        return ResultUtils.success(scheduleAttendanceService.adminUpdateAttendanceStatus(request));
    }

    /**
     * 分页查询考勤率统计。
     * 用于后台“考勤管理 / 考勤率统计”子菜单。
     */
    @PostMapping("/admin/stat/rate/page")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Page<ScheduleAttendanceRateVO>> listAttendanceRatePage(@RequestBody ScheduleAttendanceQueryRequest request) {
        ThrowUtils.throwIf(request == null, ErrorCode.PARAMS_ERROR);
        return ResultUtils.success(scheduleAttendanceService.listAttendanceRatePage(request));
    }
}
